fpga.chart.py

```diff
@@ -194,25 +185,41 @@ def _get_xilinx_fpga_power(self, idx):
 
 
     def _get_fpga_temp(self):
+        lag = 0
         while True:
             before = time.time()
             for idx in range(self.fpga_count):
-                self.temp[idx] = self.get_fpga_temp[self.manufacturer](idx)
+                self.temp[idx] = self.get_fpga_temp_func(idx)
             after = time.time()
-            time_taken = after - before
+            loop_time = self.temp_power_update_interval - (after - before)
 
-            time.sleep(self.temp_power_update_interval - time_taken)
+            if loop_time >= 0:
+                time.sleep(loop_time)
+            else:
+                lag += abs(loop_time)
+
+            if lag > self.temp_power_update_interval:
+                time.sleep(lag)
+                lag = 0
```

Comment:
Can IMO be reduced to:

    time.sleep(self.temp_power_update_interval + loop_time)